app/src/main/kotlin/com/wire/android/ui/authentication/devices/register/RegisterDevice.kt

```diff
@@ -0,0 +1,121 @@
+fun RegisterDeviceScreen() {
+    val viewModel: RegisterDeviceViewModel = hiltViewModel()
+    RegisterDeviceContent(
+        state = viewModel.state,
+        onPasswordChange = viewModel::onPasswordChange,
+        onContinuePressed = viewModel::onContinue,
+        onErrorDismiss = viewModel::onErrorDismiss
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalComposeUiApi::class)
+@Composable
+private fun RegisterDeviceContent(
+    state: RegisterDeviceState,
+    onPasswordChange: (TextFieldValue) -> Unit,
+    onContinuePressed: () -> Unit,
+    onErrorDismiss: () -> Unit
+) {
+    Scaffold(
+        topBar = {
+            WireCenterAlignedTopAppBar(
+                elevation = 0.dp,
+                title = stringResource(id = R.string.register_device_title),
+                navigationIconType = null
+            )
+        },
+    ) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Top
+        ) {
+            Text(
+                text = stringResource(id = R.string.register_device_text),
+                style = MaterialTheme.wireTypography.body01,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(
+                        horizontal = MaterialTheme.wireDimensions.spacing16x,
+                        vertical = MaterialTheme.wireDimensions.spacing24x
+                    )
+            )
+            PasswordTextField(state = state, onPasswordChange = onPasswordChange)
+            Spacer(modifier = Modifier.weight(1f))
+            WirePrimaryButton(
+                text = stringResource(R.string.label_register),
+                onClick = onContinuePressed,
+                fillMaxWidth = true,
+                loading = state.loading,
+                state = if (state.continueEnabled) WireButtonState.Default else WireButtonState.Disabled,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(MaterialTheme.wireDimensions.spacing16x)
+            )
+        }
+    }
+    if (state.error is RegisterDeviceError.GenericError)
+        CoreFailureErrorDialog(state.error.coreFailure, onErrorDismiss)
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun PasswordTextField(state: RegisterDeviceState, onPasswordChange: (TextFieldValue) -> Unit) {
+    val keyboardController = LocalSoftwareKeyboardController.current
+    WirePasswordTextField(
+            value = state.password,
+            onValueChange = onPasswordChange,
+            state = when(state.error) {
+                is RegisterDeviceError.InvalidCredentialsError ->
+                    WireTextFieldState.Error(stringResource(id = R.string.remove_device_invalid_password))
+                else -> WireTextFieldState.Default
+            },
+            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Email, imeAction = ImeAction.Done),
```

Comment:
```suggestion
            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password, imeAction = ImeAction.Done),
```
Since the type will always be Password, can it be moved to  `WirePasswordTextField`

Reply:
You're right, I completely missed it, now for `WirePasswordTextField` only `imeAction` can be set because other params shouldn't be modified